libp2p/wire.nim

```diff
@@ -64,10 +64,7 @@ proc connect*(ma: MultiAddress, bufferSize = DefaultStreamBufferSize,
   let address = initTAddress(ma)
   if address.family in {AddressFamily.IPv4, AddressFamily.IPv6}:
     if ma[1].protoCode() != multiCodec("tcp"):
-      var retFuture = newFuture[StreamTransport]()
-      retFuture.fail(newException(TransportAddressError,
-                                  "Incorrect address type!"))
-      return retFuture
+      raise newException(TransportAddressError, "Incorrect address type!")
```

Comment:
this looks wrong - if there's no {.async.} transformation, it shouldn't be raising - it should use `future.fail`

Reply:
Ah good point, that was the reason.. 
@cheatfate always recommends to use the transformation lately, should I add async and await connect?

btw the `createStreamServer` throws too (it's not async), maybe those should actually use `Result`

Reply:
`createStreamServer` is not async, so it raises.

Reply:
how about `connect` tho? transformation is preferred rather then returning plain future right?